devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,360 @@
+	if(!childDevices) {
+		def numberOfChild = getNumberOfChildFromModel()
+		log.debug "MultiChannelEndPointReport: numberOfChild '${numberOfChild}'"
+		if (numberOfChild) {
+			addChildSwitches(numberOfChild)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+def childOn(deviceNetworkId) {
+	log.info "childOn called: deviceNetworkId '${deviceNetworkId}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0xFF, switchId)
+}
+
+def childOff(deviceNetworkId) {
+	log.info "childOff called: deviceNetworkId '${deviceNetworkId}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0x00, switchId)
+}
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+
+	def cmdEncap = []
+	if (endpoint) {
+		cmdEncap = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmdEncap = cmd
+	}
+
+	if (zwaveInfo?.zw?.contains("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmdEncap).format()
+	} else if (zwaveInfo?.cc?.contains("56")){
+		zwave.crc16EncapV1.crc16Encap().encapsulate(cmdEncap).format()
+	} else {
+		cmdEncap.format()
+	}
+	
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		log.debug "changeSwitch: endpoint '${endpoint}', value: '${value}')"
+		result << child.sendEvent(name: "switch", value: value)
+		log.debug "changeSwitch: result '${result}'"
+	}
+	result
+}
+
+private getNumberOfChildFromModel() {
+	if (zwaveInfo.prod.equals("0063")) {
+		return 3
+	} else if (zwaveInfo.prod.equals("0062")) {
+		return 2
+	} else if (zwaveInfo.prod.equals("0061")) {
+		return 1
+	} else {
+		return 0
+	}
+	return 0
+}
+
+private onOffCmd(value, endpoint) {
+	log.info "onOffCmd called: val:${value}, ep:${endpoint}"
+	secureEncap(zwave.basicV1.basicSet(value: value), endpoint)
+}
+
+private getSwitchId(deviceNetworkId) {
+	log.info "getSwitchId called: ${deviceNetworkId}"
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private addChildSwitches(numberOfSwitches) {
+	log.info "addChildSwitches called: numberOfSwitches '${numberOfSwitches}'"
+	for(def endpoint : 1..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = "Dawon Smart Switch${endpoint}"
+			def child = addChildDevice("Child Switch", childDni, device.hubId,
+				[completedSetup: true, label: componentLabel, isComponent: false])
```

Comment:
Can you share a picture of this device on the Jira ticket? I want to know whether or not the child devices should be in the main device page or not.

Reply:
I have upload the product picture to the below tickets.

WWST-4892 : In development - [Powermanager] In-Wall Switch (1 gang)
WWST-4872 : In development - [Powermanager] In-Wall Switch (2 gang)
WWST-4876 : In development - [Powermanager] In-Wall Switch (3 gang)